docs/core/tools/project-json-to-csproj.md

```diff
@@ -0,0 +1,592 @@
+    "precompile": "generateCode.cmd",
+    "postpublish": [ "obfuscate.cmd", "removeTempFiles.cmd" ]
+  }
+}
+```
+
+Their equivalent in MSBuild are targets:
+
+```xml
+<Target Name="MyPreCompileTarget" BeforeTargets="Build">
+  <Exec Command="generateCode.cmd" />
+</Target>
+
+<Target Name="MyPostCompileTarget" AfterTargets="Publish">
+  <Exec Command="obfuscate.cmd" />
+  <Exec Command="removeTempFiles.cmd" />
+</Target>
+```
+
+
+## runtimeOptions
+
+```json
+{
+  "runtimeOptions": {
+    "configProperties": {
+      "System.GC.Server": true
+    }
+  }
+}
+```
+
+All settings in this group should be placed in the project folder into a file called
+*runtimeconfig.template.json*, with options lifted to the root object, 
+like in the following example:
+
+```json
+{
+  "configProperties": {
+    "System.GC.Server": true
+  }
+}
+```
+
+## shared
+```json
+{
+  "shared": "shared/**/*.cs"
+}
+```
+
+Not supported in csproj. You must instead create include content files in your *.nuspec* file. 
+For more information, see [Including content files](https://docs.microsoft.com/nuget/schema/nuspec#including-content-files).
+
+## files
+
+In *project.json*, build and pack could be extended to compile and embed from different folders.
+In MSBuild, this is done using items. The following example is a common conversion:
+
+```json
+{
+  "buildOptions": {
+    "compile": {
+      "copyToOutput": "notes.txt",
+      "include": "../Shared/*.cs",
+      "exclude": "../Shared/Not/*.cs"
+    },
+    "embed": {
+      "include": "../Shared/*.resx"
+    }
+  },
+  "packOptions": {
+    "include": "Views/",
+    "mappings": {
+      "some/path/in/project.txt": "in/package.txt"
+    }
+  },
+  "publishOptions": {
+    "include": [
+      "files/",
+      "publishnotes.txt"
+    ]
+  }
+}
+```
+
+```xml
+<ItemGroup>
+  <Compile Include="..\Shared\*.cs" Exclude="..\Shared\Not\*.cs" />
+  <EmbeddedResource Include="..\Shared\*.resx" />
+  <Content Include="Views\**\*" PackagePath="%(Identity)" />
+  <None Include="some/path/in/project.txt" Pack="true" PackagePath="in/package.txt" />
+  
+  <None Include="notes.txt" CopyToOutputDirectory="Always" />
+  <!-- CopyToOutputDirectory = { Always, PreserveNewest, Never } -->
+
+  <Content Include="files\**\*" CopyToPublishDirectory="PreserveNewest" />
+  <None Include="publishnotes.txt" CopyToPublishDirectory="Always" />
+  <!-- CopyToPublishDirectory = { Always, PreserveNewest, Never } -->
+</ItemGroup>
+```
+
+> [!NOTE]
+> Many of default globbing patterns are added automatically by the .NET Core SDK.
+> For more information, see [Default Compile Item Values](https://aka.ms/sdkimplicititems).
+
+All MSBuild `ItemGroup` elements support `Include`, `Exclude`, and `Remove`.
+
+Package layout inside the .nupkg can be modified with `PackagePath="path"`.
+
+Except for `Content`, most item groups require explicitly adding `Pack="true"` to 
+be included in the package. By default, this will be but in the 'content' folder
```

Comment:
Might be worth mentioning that ```IncludeContentInPack```, which is true by default, is what makes all items of type ```Content``` to be added to the nupkg by default. This behavior can be overridden by setting it  to false.